src/index.ts

```diff
@@ -697,39 +688,49 @@ export async function startAnalysis<
                     if (targetOptions?.managed || targetOptions?.startEmulatorOptions) {
                         // Create or start a new emulator
                         // eslint-disable-next-line require-atomic-updates
-                        emulator = await AndroidEmulator.fromRunTarget(targetOptions);
+                        emulator = await AndroidEmulator.fromRunTarget(targetOptions, analysisOptions.capabilities);
                         await emulator.start();
                     }
                 }
 
                 const abortSignal = emulator?.getAbortSignal();
                 abortSignal?.throwIfAborted();
 
-                await Promise.race([
-                    (async () => {
-                        await platform.waitForDevice(150, { abortSignal });
-
-                        if (emulator?.resetSnapshotName)
-                            await timeout(platform.resetDevice(emulator?.resetSnapshotName, { abortSignal }), {
-                                milliseconds: 5 * 60 * 1000,
-                                signal: abortSignal,
-                            });
-
-                        await platform.ensureDevice({ abortSignal });
+                await platform.waitForDevice(150, { abortSignal });
+
+                if (emulator?.resetSnapshotName)
+                    await timeout(platform.resetDevice(emulator?.resetSnapshotName, { abortSignal }), {
+                        milliseconds: 5 * 60 * 1000,
+                        signal: abortSignal,
+                    });
+
+                await platform.ensureDevice({ abortSignal });
+
+                if (
+                    !emulator?.resetSnapshotName &&
+                    targetOptions?.managed &&
+                    (analysisOptions?.capabilities as unknown as SupportedCapability<'android'> | undefined)?.includes(
+                        'frida'
```

Comment:
We should error if `honeyData` is specified but `frida` isn't in the `capabilities`.